testsuites/sparql/src/main/java/org/eclipse/rdf4j/testsuite/sparql/tests/BindTest.java

```diff
@@ -266,4 +270,14 @@ public void testGH4499BindFilterNotExist2() {
 		assertThat(bs.getValue("d")).isNull();
 
 	}
+
+	public Stream<DynamicTest> tests() {
+		return Stream.of(makeTest("GH4499BindFilterNotExist2", this::testGH4499BindFilterNotExist2),
+				makeTest("GH4499BindFilterNotExist1", this::testGH4499BindFilterNotExist1),
+				makeTest("GH3696Bind", this::testGH3696Bind), makeTest("SelectBindOnly", this::testSelectBindOnly),
+				makeTest("SES2250BindErrorsInPath", this::testSES2250BindErrorsInPath),
+				makeTest("SES2250BindErrors", this::testSES2250BindErrors),
+				makeTest("BindScopeUnion", this::testBindScopeUnion), makeTest("BindScope", this::testBindScope),
+				makeTest("BindError", this::testBindError));
+	}
```

Comment:
Is there no way to stick with the regular @Test annotation? Or maybe generate this list of tests in a more automatic way?

Reply:
I couldn't figure it out from the documentation :(

Reply:
ChatGPT recommends this:

```java
import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;
import java.util.Arrays;
import java.util.stream.Stream;

import org.junit.jupiter.api.DynamicTest;
import org.junit.jupiter.api.function.Executable;

public class BindTest extends AbstractComplianceTest {

    // ... existing methods ...

    public Stream<DynamicTest> tests() {
        return Arrays.stream(BindTest.class.getDeclaredMethods())
                     .filter(method -> shouldRunAsTest(method))
                     .map(this::createDynamicTestFromMethod);
    }

    private boolean shouldRunAsTest(Method method) {
        // Use your own logic here to determine which methods are test methods
        // For example, you might check for certain annotations or naming patterns
        return method.getName().startsWith("test");
    }

    private DynamicTest createDynamicTestFromMethod(Method method) {
        return DynamicTest.dynamicTest(method.getName(), new Executable() {
            @Override
            public void execute() throws Throwable {
                try {
                    method.invoke(BindTest.this);
                } catch (IllegalAccessException | InvocationTargetException e) {
                    throw e.getCause();
                }
            }
        });
    }
}
```

Reply:
Dunno how pretty that is though.

Reply:
That's going back to junit v2 ;) I don't mind, the official way is to generate some kind of test template. But it is a lot of work for a little gain. @hmottestad Is it ok if we merge as is and leave that as a todo?

Reply:
I guess we can always change it up later, and your approach does allow IntelliJ to run a single test instead of having to run the entire suite.